tests/bfgs-strategy.cpp

```diff
@@ -0,0 +1,134 @@
+#include <boost/test/unit_test.hpp>
+#include "util.hpp"
+
+#include "proxsuite-nlp/bfgs-strategy.hpp"
+
+using namespace proxsuite::nlp;
+
+BOOST_AUTO_TEST_CASE(test_inverse_hessian_update) {
```

Comment:
I don't understand this test. Can you comment what it supposed to do ?

This comment apply to all the tests.